### ISSUE TITLE
frankie "whyengineer" fork of ARM's CMSIS for ESP32

### ISSUE THREAD

**Issue by StuartIanNaylor** [atomic14/diy-alexa#7]
Have no idea but soon as I saw it I thought oh that is interesting as https://github.com/UT2UH/ML-KWS-for-ESP32 is just an implementation of https://github.com/ARM-software/ML-KWS-for-MCU

Close off my 'issues' as they are really not just wondered like the beamforming stereo mic and using esp32s as a distributed array might be ideas of interest.

Thought I would post another and might be outdated now as this is the old 1.15 version of tensorflow but if you look at Accuracy of the models on validation set, their memory requirements and operations per inference in the table of the above.

The CRNN and DS-CNN architectures are really interesting and might be far better than a plain CNN if frankie "whyengineer" fork of ARM's CMSIS for ESP32 works as that is a collection of Arm boffin fast math where we don't have native libs to make the above examples work.

Maybe it not the fast math and just the driver pack but you will know with a faster glance than I or substitute math libs that do similar?

**Comment by StuartIanNaylor** [comment]
Also another thing Chris and I am not sure of this as different sources all seem to have different settings for stride and frame size.
https://haythamfayek.com/2016/04/21/speech-processing-for-machine-learning.html

> Typical frame sizes in speech processing range from 20 ms to 40 ms with 50% (+/-10%) overlap between consecutive frames. Popular settings are 25 ms for the frame size, frame_size = 0.025 and a 10 ms stride (15 ms overlap), frame_stride = 0.01

You are always going to get difference due to sampling rate and changes in duration and its prob preference down to language as phonetic speed can differ.

But what puzzles me is some say have an overlap and some examples seem to omit, Librosa actually is 4:1 frame stride as think they allow a half window overlap.

Its something I haven't managed to work out which is correct?

Also Linto HMG the Master version is older but they have done some updates to version 1.0 but haven't checked yet.
